### PR TITLE
Add iotivity_1.2-rel

### PR DESCRIPTION
**Pull request by heejin-kim**
Please review @sangwon03 @lssgood @pillip8282 @HONGCHAEHEE @sunghan-chang

### CODE REVIEW

**Comment by sunghan-chang** [comment]
You must fix violations of Tizen coding rule except extern folder.

File: apps/examples/iotivity_simpleserver/Makefile [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+############################################################################
+############################################################################
+# apps/examples/iotivity/Makefile
```
Comment (arpit-IITG14) [inline apps/examples/iotivity_simpleserver/Makefile]: I think it should be apps/examples/iotivity_simpleserver/Makefile

**Comment by JunhwanPark** [comment]
Hi @heejin-kim 
I am testing your patch.
Should Iotivity in TizenRT be compiled with Scon?
The existing TizenRT uses the Makefile.

File: apps/examples/iotivity_simpleserver/Makefile (hunk 1 above)
Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/Makefile]: I see.

**Comment by heejin-kim** [comment]
@sunghan-chang Ok, I will amend that.
@arpit-IITG14 I agree with you.
@JunhwanPark You're right. Tizen RT uses the Makefile to build. But IoTivity, in the external folder, should use Scon to follow their own build policy.

**Comment by JunhwanPark** [comment]
Hi @heejin-kim 
I know that Iotivity uses Scons. However, when you ported Iotivity to TizenRT, I did not know if it was necessary to use Scons, so I asked you.
Thanks for the reply. :+1: 
Do you know if you can build with Scons and debug with GDB or Trace32?

File: apps/examples/iotivity_simpleserver/mkfsdata.pl [hunk 2]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/perl
+###########################################################################
+#
+# Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: 2016 -> 2017

File: apps/examples/iotivity_simpleserver/mkfsdata.pl [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+# tools/mkfsdata.pl
+#
+# Extracted from uIP which has a license that is compatible with NuttX.
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: Is this come from NuttX?

File: apps/examples/iotivity_simpleserver/ocserver.h [hunk 4]
```diff
@@ -0,0 +1,131 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/ocserver.h]: 2017

File: apps/examples/iotivity_simpleserver/port_tinyara.c [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+#include <sys/time.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <net/lwip/netif.h>
+#include <ocstack.h>
+#include "uuid/uuid.h"
+
+#if CONFIG_NSOCKET_DESCRIPTORS > 0
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/port_tinyara.c]: You need including <tinyara/configh.> for this .

File: apps/examples/iotivity_simpleserver/simpleServer.c [hunk 6]
```diff
@@ -0,0 +1,1078 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/simpleServer.c]: 2017

File: apps/examples/iotivity_simpleserver/mkfsdata.pl (hunk 2 above)
Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: ok

File: apps/examples/iotivity_simpleserver/mkfsdata.pl (hunk 3 above)
Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: It comes from IoTivity.

Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: It looks NuttX's license. I want to know why it has NuttX's license even though it is not come from NuttX.

Comment (arpit-IITG14) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: Nuutx license can be removed from this file. I think this file was newly written to copy iotivity cert / dat file data into tinyara file system at run time.

Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/mkfsdata.pl]: I see. I will remove that license.

File: apps/examples/iotivity_simpleserver/port_tinyara.c (hunk 5 above)
Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/port_tinyara.c]: Ok. Thanks

File: apps/examples/iotivity_simpleserver/Makefile [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+install: $(BIN_DIR)$(DELIM)$(PROGNAME)
+
+else
+install:
+
+endif
+
+ifeq ($(CONFIG_BUILTIN_APPS)$(CONFIG_EXAMPLES_IOTIVITY),yy)
+$(BUILTIN_REGISTRY)$(DELIM)$(APPNAME)_main.bdat: $(DEPCONFIG) Makefile
+	$(call REGISTER,$(APPNAME),$(APPNAME)_main,$(THREADEXEC))
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/Makefile]: I can not find simpleServer_main function. Do you want to register this as TASH command? Do you know whtat this is?

Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/Makefile]: There is in simpleServer.c at the bottom. Also I already confirmed and tested that function using TASH command.
When the function called, create task, call simpleServer_cb, and so on.

Comment (lssgood) [inline apps/examples/iotivity_simpleserver/Makefile]: There exist simplesersver_main() in line 1099.

Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/Makefile]: Ok. it is hidden. Thank you.
But it makes another task to launch iotivity simple server. 
Do you need to make one more task? Is it a reason that you want to set specific stack size and priority?
Currently, TASH makes a task for ASYNC. but that makes another task again.

Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/Makefile]: I see. I will revise that. Thanks.

File: apps/examples/iotivity_simpleserver/simpleServer.c [hunk 8]
```diff
@@ -0,0 +1,1012 @@
...
+#include <arpa/inet.h>
+#include <apps/netutils/netlib.h>
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+
+#define IFHWADDRLEN 6
+
+#if defined(LWIP_HAVE_LOOPIF) && LWIP_HAVE_LOOPIF == 1
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/simpleServer.c]: Why do you compare LWIP_HAVE_LOOPIF with 1?
Is it possible that LWIP_HAVE_LOOPIF has another value?

Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/simpleServer.c]: When LWIP_HAVE_LOOPIF is not defined, this is defined with 0 in os/include/net/lwip/opt.h. So, this is always defined but can have different value. Let's remove checking definition.
#if LWIP_HAVE_LOOP == 1

Comment (heejin-kim) [inline apps/examples/iotivity_simpleserver/simpleServer.c]: Honestly, I don't know why.. I searched that part of the code, there is no need to be there. I will erase that.

File: apps/examples/iotivity_simpleserver/simpleServer.c [hunk 9]
```diff
@@ -0,0 +1,1012 @@
...
+#include <arpa/inet.h>
+#include <apps/netutils/netlib.h>
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+
+#define IFHWADDRLEN 6
+
+#ifdef LWIP_HAVE_LOOPIF
```
Comment (sunghan-chang) [inline apps/examples/iotivity_simpleserver/simpleServer.c]: If you include lwipopts.h only, this is correct.
But if you include opt.h, this is wrong. You should use #if LWIP_HAVE_LOOPIF == 1.

In your codes, header file is included in-directly. It make us difficult to check what header file is included.

**Comment by sunghan-chang** [comment]
Config, EXAMPLE_IOTIVITY, to enable iotivity example is in Kconfig of extern, not in example/iotivity_simpleserver even though every codes for example are in example.

**Pull request merged**